### ISSUE TITLE
linter: jsPlugins and their rules are not inherited through `extends

### ISSUE THREAD

**Issue by jterrazz** [oxc-project/oxc#17320]
### What version of Oxlint are you using?

1.35.0

### What command did you run?

```bash
npx oxlint
```

### What does your `.oxlintrc.json` config file look like?

**Shared config** (`src/oxlint/architectures/hexagonal.json`):
```json
{
  "jsPlugins": ["./plugins/codestyle.js"],
  "rules": {
    "codestyle/arch-hexagonal": ["error", { "rules": [...] }]
  }
}
```

**User config** (`.oxlintrc.json`):
```json
{
  "extends": ["@jterrazz/codestyle/oxlint/architectures/hexagonal"]
}
```

The `jsPlugins` and `rules` from the shared config are not inherited.

### What happened?

When using `extends` to share oxlint configuration from a package, `jsPlugins` and any rules from those plugins are not inherited by the consuming project.

### Reproduction

1. Create a shared config package with:
```json
// @mypackage/oxlint/hexagonal.json
{
  "jsPlugins": ["./plugins/my-plugin.js"],
  "rules": {
    "my-plugin/my-rule": "error"
  }
}
```

2. Extend it in a project:
```json
// .oxlintrc.json
{
  "extends": ["@mypackage/oxlint/hexagonal"]
}
```

3. Run oxlint - the JS plugin is not loaded and the rule is not recognized.

### Expected behavior

`jsPlugins` and their associated rules should be inherited from extended configs, just like other config options.

### Current workaround

We use a postinstall script to patch the user's `.oxlintrc.json` with the required `jsPlugins` and rules.

### Reproducible example

This issue can be reproduced at https://github.com/jterrazz/package-codestyle:

1. Install the package in any project: `npm install @jterrazz/codestyle`
2. Create `.oxlintrc.json` with only: `{ "extends": ["@jterrazz/codestyle/oxlint/architectures/hexagonal"] }`
3. Run `npx oxlint` - the `codestyle/arch-hexagonal` rule from the extended config's `jsPlugins` is not loaded

The package currently works around this with a postinstall script that patches the user's config.

### Environment

- oxlint version: 1.35.0
- OS: macOS

**Comment by gaurav-rk9** [comment]
`jsPlugin` is inherited from `extends`, but `extends` does not support npm module resolution.
We can see that by replacing `extends` field to `["node_modules/@jterrazz/codestyle/src/oxlint/architectures/hexagonal.json"]`.

relevant code - https://github.com/oxc-project/oxc/blob/585ddf80ac5a106d2ebc93ae786c4592c6546d36/crates/oxc_linter/src/config/config_builder.rs#L129

**Comment by jterrazz** [comment]
Ok it is indeed working, thanks !

**Comment by connorshea** [comment]
I'm not sure that it's intentional for these to not be inherited through extends? So this issue should maybe stay open.

**Comment by rahdirigs** [comment]
Can we consider supporting npm module resolution? This would be provide for a much cleaner way to inherit from presets like ultracite or even through a shared `@myorg/oxlint-config`package within a monorepo.

**Comment by patricktree** [comment]
cross-referencing issue https://github.com/oxc-project/oxc/issues/23944 which would fix the missing node_modules resolution of js plugins
